src/renderer/viz/expressions/ramp.js

```diff
@@ -325,6 +342,35 @@ export default class Ramp extends BaseExpression {
         gl.uniform1f(this._getBinding(program).keyMinLoc, (this.minKey));
         gl.uniform1f(this._getBinding(program).keyWidthLoc, (this.maxKey) - (this.minKey));
         drawMetadata.freeTexUnit++;
+
+        if (this.input.type === 'category' && this.input.isA(Property)) {
+            gl.activeTexture(gl.TEXTURE0 + drawMetadata.freeTexUnit);
+            if (this._translatedIds !== this._metadata.properties[this.input.name].categories.length) {
+                this._translatedIds = this._metadata.properties[this.input.name].categories.length;
+                this._translateTexture = gl.createTexture();
+                const translatorPixels = new Float32Array(256 * 256);
+                for (let i = 0; i < this._metadata.properties[this.input.name].categories.length; i++) {
+                    const id = this._metadata.categoryToID.get(this._metadata.properties[this.input.name].categories[i].name);
```

Comment:
Just as an improvement, I would add a method to the `Metadata` class to get the property categories:

```js
    getPropertyCategories (propertyName) {
        return this._metadata.properties[propertyName].categories;
    }
```

So the loop would look like:

```js
const translatorPixels = new Float32Array(COLOR_ARRAY_LENGTH * COLOR_ARRAY_LENGTH);
const propertyCategories = this.metadata.getPropertyCategories(this.input.name);
const PROPERTY_CATEGORY_LENGTH = propertyCategories.length;

for (let i = 0; i < PROPERTY_CATEGORY_LENGTH; i++) {
    const id = propertyCategories[i].id;
    const value = i / (PROPERTY_CATEGORY_LENGTH - 1);
    const vec2Id = {
        x: id % COLOR_ARRAY_LENGTH,
        y: Math.floor(id / COLOR_ARRAY_LENGTH)
    };
    translatorPixels[COLOR_ARRAY_LENGTH * vec2Id.y + vec2Id.x] = value;
}
```